src/libstd/sys/unix/fd.rs

```diff
@@ -50,14 +50,14 @@ impl FileDesc {
         Ok(ret as usize)
     }
 
-    #[cfg(not(target_env = "newlib"))]
+    #[cfg(not(any(target_env = "newlib", target_os = "sunos")))]
```

Comment:
It does, but `FIOCLEX` apparently doesn't work there (although it's available). So the solution is to set the cloexex flag by the means of `fcntl`.
I figured it out through some indirect references, e.g.: https://forum.nginx.org/read.php?29,232210,232210 (notice the Solaris workaround).